src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +432,95 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
```

Comment:
I see, so your intention isn't to have timer-based polling, but rather to have this function sit in a blocking loop as long as there's work to do?

If so, it's strange that this is an async method.  Note that having this be "async" doesn't mean it's invoked asynchronously... just means that it's allowed to wait.  So I'm surprised to see the comment below that it's rare to hit that Task.Delay code path... seems like you'd hit that every time, as otherwise this function will block its caller (AddEasyHandle), no?

If we're going with this thread-blocked-per-CurlHandler scheme (something I'm still not clear on why we need to nor particularly happy about), I'd expect:
- We'd schedule an invocation of this function via Task.Run when the count of FDs increases from 0 to 1.
- We'd exit the loop when the count drops back to 0.